websites/D/Doronime/presence.ts

```diff
@@ -8,96 +8,119 @@ const presence = new Presence({
+						? "Viewing Anime Info"
+						: title;
+				presenceData.state =
+					document.location.pathname.slice(1).split("/").length === 2
+						? title
+						: capitalize(
+								document.location.pathname
+									.slice(1)
+									.split("/")[2]
+									.replace("bd-", "")
+									.split("-")
+						  );
+			} else presenceData.details = "Viewing Anime List";
+			break;
+		case "ost":
+			if (data.length) {
+				const [title, artist] = [
+					...document.querySelectorAll(
+						".Content__description-caption > .form-row > div"
+					),
 				]
-					.slice(1)
-					.map(x => {
-						if (x.querySelector("a")) return x.querySelector("a").textContent;
-						return x.querySelector("span").textContent;
-					}),
-				parseDetails = animeInfo[0].includes("List")
-					? "Viewing Anime List"
-					: animeInfo[1].includes("Batch")
-					? "Viewing Anime Batch"
-					: animeInfo[0],
-				parseState = animeInfo[0].includes("List")
-					? animeInfo[1]
-					: animeInfo[1].includes("Batch")
-					? animeInfo[0]
-					: animeInfo[1];
-			presenceData.details = parseDetails;
-			presenceData.state = parseState;
-			presenceData.buttons = [
-				{
-					label: "View Anime",
-					url: document.location.href,
-				},
-			];
-		} else presenceData.details = "Viewing Anime List";
-	}
-	if (base === "batch") presenceData.details = "Viewing Anime Batch List";
-	if (base === "movie") presenceData.details = "Viewing Anime Movie List";
-	if (base === "ost") {
-		if (data.length) {
-			const [title, artist] = [
-				...document.querySelectorAll(
-					".Content__description-caption > div > div > div"
-				),
-			].map(x => x.textContent);
-			presenceData.details = artist;
-			presenceData.state = title;
-			presenceData.buttons = [
-				{
-					label: "View OST",
-					url: document.location.href,
-				},
-			];
-		} else presenceData.details = "Viewing Anime OST List";
-	}
-	if (base === "genre") {
-		if (data.length) {
-			presenceData.details = `Viewing Anime Genre ${
-				document.querySelector(".Content__tabs > div > span").textContent
-			}`;
-		} else presenceData.details = "Viewing Anime Genre List";
-	}
-	if (base === "season") {
-		if (data.length) {
-			presenceData.details = `Viewing Anime Season ${
-				document.querySelector(".Content__tabs > div > span").textContent
-			}`;
-		} else presenceData.details = "Viewing Anime Season List";
+					.map((x, i) => (i % 2 === 1 ? x : null))
+					.filter(Boolean)
+					.splice(0, 2)
+					.map(x =>
+						x.textContent.replace(/ +?/g, "").split("\n").join(" ").trim()
+					);
+				presenceData.details =
+					title.split(" ").length > 1 ? title.split(" ")[1] : title;
+				presenceData.state = artist;
+			} else presenceData.details = "Viewing Anime OST List";
+			break;
+		case "genre":
+			if (data.length) {
+				presenceData.details = "Viewing Anime Genre";
+				presenceData.state = document.querySelector(
+					".Content__tabs > div > span"
+				).textContent;
```

Comment:
![image](https://github.com/PreMiD/Presences/assets/42322979/93e23a4a-0d6b-4975-a79b-9bc2c1bf5525)
```suggestion
				presenceData.state = trim(document.querySelector(
					'[aria-current="page"]'
				)?:.textContent);
```